src/appdistribution/client.ts

```diff
@@ -1,182 +1,158 @@
-  SUCCESS = "SUCCESS",
-  IN_PROGRESS = "IN_PROGRESS",
-  ERROR = "ERROR",
+export interface TestCertificate {
+  hashSha1: string;
+  hashSha256: string;
+  hashMd5: string;
 }
 
 /** Enum representing the App Bundles state for the App */
-export enum AabState {
-  AAB_STATE_UNSPECIFIED = "AAB_STATE_UNSPECIFIED",
-  ACTIVE = "ACTIVE",
+export enum IntegrationState {
+  AAB_INTEGRATION_STATE_UNSPECIFIED = "AAB_INTEGRATION_STATE_UNSPECIFIED",
+  INTEGRATED = "INTEGRATED",
   PLAY_ACCOUNT_NOT_LINKED = "PLAY_ACCOUNT_NOT_LINKED",
   NO_APP_WITH_GIVEN_BUNDLE_ID_IN_PLAY_ACCOUNT = "NO_APP_WITH_GIVEN_BUNDLE_ID_IN_PLAY_ACCOUNT",
   APP_NOT_PUBLISHED = "APP_NOT_PUBLISHED",
   AAB_STATE_UNAVAILABLE = "AAB_STATE_UNAVAILABLE",
   PLAY_IAS_TERMS_NOT_ACCEPTED = "PLAY_IAS_TERMS_NOT_ACCEPTED",
 }
 
-export interface UploadStatusResponse {
-  status: UploadStatus;
-  message: string;
-  errorCode: string;
-  release: {
-    id: string;
-  };
+export enum UploadReleaseResult {
+  UPLOAD_RELEASE_RESULT_UNSPECIFIED = "UPLOAD_RELEASE_RESULT_UNSPECIFIED",
+  RELEASE_CREATED = "RELEASE_CREATED",
+  RELEASE_UPDATED = "RELEASE_UPDATED",
+  RELEASE_UNMODIFIED = "RELEASE_UNMODIFIED",
+}
+
+export interface Release {
+  name: string;
+  releaseNotes: ReleaseNotes;
+  displayVersion: string;
+  buildVersion: string;
+  createTime: Date;
+}
+
+export interface ReleaseNotes {
+  text: string;
+}
+
+export interface UploadReleaseResponse {
+  result: UploadReleaseResult;
+  release: Release;
 }
 
-/** Enum for app_view parameter for getApp requests */
-export enum AppView {
-  BASIC = "BASIC",
-  FULL = "FULL",
+export interface BatchRemoveTestersResponse {
+  emails: string[];
 }
 
 /**
- * Proxies HTTPS requests to the App Distribution server backend.
+ * Makes RPCs to the App Distribution server backend.
  */
 export class AppDistributionClient {
-  static MAX_POLLING_RETRIES = 60;
-  static POLLING_INTERVAL_MS = 2000;
+  appDistroV2Client = new Client({
+    urlPrefix: api.appDistributionOrigin,
+    apiVersion: "v1",
+  });
 
-  constructor(private readonly appId: string) {}
-
-  async getApp(appView = AppView.BASIC): Promise<AppDistributionApp> {
-    const apiResponse = await api.request("GET", `/v1alpha/apps/${this.appId}?appView=${appView}`, {
+  async getAabInfo(appName: string): Promise<AabInfo> {
+    const apiResponse = await api.request("GET", `/v1/${appName}/aabInfo`, {
       origin: api.appDistributionOrigin,
       auth: true,
     });
 
     return _.get(apiResponse, "body");
   }
 
-  async uploadDistribution(distribution: Distribution): Promise<string> {
-    const apiResponse = await api.request("POST", `/app-binary-uploads?app_id=${this.appId}`, {
+  async uploadRelease(appName: string, distribution: Distribution): Promise<string> {
+    const apiResponse = await api.request("POST", `/upload/v1/${appName}/releases:upload`, {
       auth: true,
       origin: api.appDistributionOrigin,
       headers: {
-        "X-APP-DISTRO-API-CLIENT-ID": pkg.name,
```

Comment:
I think we want to keep the trio of `X-App-Distro-Api` fields since they distinguish our client ID traffic.

Reply:
No need. The Firebase CLI infrastructure [automatically adds an `X-Client-Version` header](https://github.com/firebase/firebase-tools/search?q=X-Client-Version) and [our code gets the three values it needs](http://google3/java/com/google/firebase/appdistro/boq/developers/v1/service/actions/UploadReleaseAction.java?l=206,208,260-261&rcl=394538100) - two (name and version of the client) by parsing the `X-Client-Version` header and one from the app (the platform).